src/cpp/utils/threading.hpp

```diff
@@ -29,13 +29,30 @@ struct ThreadSettings;
 /**
  * @brief Give a name to the thread calling this function.
  *
- * @param[in]  name  A null-terminated string with the name to give to the calling thread.
- *                   The implementation for certain platforms may truncate the final thread
- *                   name if there is a limit on the length of the name of a thread.
+ * @param[in, out]  thread_name_buffer  Buffer to store the name of the thread.
+ * @param[in]       name  A null-terminated string with the name to give to the calling thread.
+ *                        The implementation for certain platforms may truncate the final thread
+ *                         name if there is a limit on the length of the name of a thread.
  */
 void set_name_to_current_thread(
+        char* thread_name_buffer,
```

Comment:
Would be better to enforce a correct size of this buffer:

```suggestion
        std::array<char, 16>& thread_name_buffer,
```

Reply:
Applies to all overloads of `set_name_to_current_thread`.